linera-service/tests/linera_net_tests.rs

```diff
@@ -1009,8 +1009,9 @@ async fn test_wasm_end_to_end_non_fungible(config: impl LineraNetConfig) -> Resu
 
     let nft1_blob = Blob::test_data_blob("nft1_data");
     let nft1_blob_id = nft1_blob.id();
+    let nft1_inner_data = nft1_blob.into_inner_bytes();
```

Comment:
Should we just call it `nft1_payload` and not even construct the blob itself?

Reply:
The thing is that we also need to test if the hash being generated in the `BlobId` matches what the publish is returning. We could just create the `BlobId` from `BlobContent`, but then I guess we might as well just create the full `Blob` and get the id generated from the full `Blob` creation

Reply:
But users shouldn't have to deal with general blobs either, and they should be able to compute their data blob hashes. So we should be able to do the same in the tests. I'd use the general blob abstraction only where we actually need it.

Reply:
Sounds good! I'll do that as much as I can. However, some tests require me to manually call `add_pending_blobs`, which will require an actual `Blob` object. In those cases I'll keep using the blob abstraction